### PR TITLE
Fix work package dates for overflowing work packages in team planner

### PR DESCRIPTION
**Pull request by b12f**
Closes https://community.openproject.org/work_packages/41723/activity
Closes https://community.openproject.org/projects/openproject/work_packages/41118/activity

### CODE REVIEW

File: frontend/src/app/features/team-planner/team-planner/planner/team-planner.component.ts [hunk 1]
```diff
@@ -601,25 +601,30 @@ export class TeamPlannerComponent extends UntilDestroyedMixin implements OnInit,
...
+
+    const viewEndDate = this.ucCalendar.getApi().view.currentEnd.setHours(0, 0, 0, 0);
+    const dateToCheck = new Date(dueDate).setHours(0, 0, 0, 0);
 
-      if (date === 'start') {
-        dateToCheck = new Date(workPackage.startDate).setHours(0, 0, 0, 0);
-      } else {
-        dateToCheck = new Date(workPackage.dueDate).setHours(0, 0, 0, 0);
-      }
+    return dateToCheck < viewEndDate;
```
Comment (HDinger) [inline frontend/src/app/features/team-planner/team-planner/planner/team-planner.component.ts:614]: This should be `=<`

File: frontend/src/app/features/work-packages/components/wp-card-view/wp-single-card/wp-single-card.component.html [hunk 2]
```diff
@@ -81,27 +81,30 @@
     <span class="op-wp-single-card--content-subject-line">
       <span
         *ngIf="showAsInlineCard && showStartDate"
-        class="op-wp-single-card--content-inline-date">
-        {{splittedDate(workPackage)[0]}}-
+        class="op-wp-single-card--content-inline-date"
+      >
+        {{startDate(workPackage)}}
```
Comment (HDinger) [inline frontend/src/app/features/work-packages/components/wp-card-view/wp-single-card/wp-single-card.component.html]: The dash is missing here (e.g. "12. Apr -"). Same for the end date.

**Review by HDinger — COMMENTED** [review]
Looks good as far as I can tell. I only have two minor remarks 👍

File: frontend/src/app/features/team-planner/team-planner/planner/team-planner.component.ts (hunk 1 above)
Comment (b12f) [reply frontend/src/app/features/team-planner/team-planner/planner/team-planner.component.ts:614]: No, I'm not sure why exactly, but the `viewEndDate` is the date past the end of the calendar.

**Review by HDinger — APPROVED** [review]

**Comment by HDinger** [comment]
There is one eslint issue left, other than that it is good to merge 👍

**Pull request merged**